python/pyarrow/_dataset.pyx

```diff
@@ -1963,6 +1963,100 @@ cdef class FragmentScanOptions(_Weakrefable):
         except TypeError:
             return False
 
+cdef class CacheOptions(_Weakrefable):
+    """
+    Cache options for a pre-buffered fragment scan.
+    Parameters
+    ----------
+    hole_size_limit : int, default 8Ki
+        The maximum distance in bytes between two consecutive ranges; beyond 
+        this value, ranges are not combined.
+    range_size_limit : int, default 32Mi
+        The maximum size in bytes of a combined range; if combining two 
+        consecutive ranges would produce a range of a size greater than this, 
+        they are not combined
+    lazy : bool, default False
+        A lazy cache does not perform any I/O until requested.
```

Comment:
Hmmm this is a bit confusing with `prefetch_limit`? (Although the C++ comment has the same problem)

Reply:
Probably I just copied the C++ comment. Its been so long I can't actually remember what this does. I will read the code a bit and try to improve the comments.

Reply:
Thanks @mapleFU. I altered the wording a bit but I've updated the comments on the python and C++.